graph_dfs_debug/graph.py

```diff
@@ -19,36 +19,40 @@ def add_vertex(self, vertex, edges=()):
         self.vertices[vertex] = set(edges)
 
     def add_edge(self, start, end, bidirectional=True):
-        self.vertices[start].add(start)
+        self.vertices[start].add(end)
         if bidirectional:
             self.vertices[end].add(end)
 
     def dfs(self, start, target=None):
-        x = []
-        x.append(start)
-        y = set(x)
+        stack = []
+        stack.append(start)
+        visited = set()
 
-        while x:
-            z = x.pop()
-            if x == target:
-                break
-            x.extend(self.vertices[z])
+        while stack:
+            current = stack.pop()
+            if current == target:
+                # print("Target {}".format(target))
+                return target
+            visited.add(current)
+            stack.extend(self.vertices[current] - visited)
+            
```

Comment:
if current == target would you not want to 'break' out of the loop? Also..where are you updating the visited set?